### PR TITLE
Flight profile command

### PR DESCRIPTION
**Pull request by shorstman**
Added flight profile commands `fp` and `fpget`. `fp`is for setting values in the flight profile, `fpget` returns all values in the flight profile. Also added documentation for the new commands in the COMMANDS.md file.

### CODE REVIEW

**Comment by shorstman** [comment]
Somehow every file got committed during my last commit. Not sure why that happened.

**Comment by EdHurtig** [comment]
mmm sounds like you have a BUILD directory in a non-ignored location...
`git rm -rf ./BUILD`

**Comment by EdHurtig** [comment]
Ooooo I see what happened here...

You ran `cmake` from the root directory.  That `./proj` folder is specifically placed in the repo for the `cmake` artifacts and is ignored by default.

**Comment by shorstman** [comment]
What should I do to fix it?

**Comment by EdHurtig** [comment]
Just go and remove each of the files/folders that shouldn't be there with `git rm -rf <badfile>`, then `git commit -m "Fixed the badfiles"` and then push.

**Comment by EdHurtig** [comment]
Alternatively: If you don't want a extra commit you can do the `git rm`'s and then `git commit --amend` and then force push with `git push -f origin flight-profile-command`

**Comment by shorstman** [comment]
When you say all the files that shouldn't be there does that just mean any files I didn't edit on this branch?

**Comment by EdHurtig** [comment]
alternatively:

```
git reset --soft HEAD^ # Remove the previous commit, but keep the changes
git reset # Unstage all files
git status # Look at the status now
git add path/to/file/i/changed.c
git commit -m "Something"
git push -f origin flight-profile-command
```

SO MANY OPTIONS

**Comment by EdHurtig** [comment]
Yeah by "all the files that shouldn't be there" I mean the Cmake files and Build artifacts and all that jazz. The files that were new but not added by you

**Comment by EdHurtig** [comment]
Yep that did it!

File: core/commands.c [hunk 1]
```diff
@@ -328,9 +329,107 @@ static int killCommand(size_t argc, char *argv[], size_t outbufc, char outbuf[])
...
+                                    profile->watchdog_timer, profile->emergency_hold,
+                                    profile->braking_wait, profile->pusher_timeout,
+                                    profile->pusher_state_accel_min, profile->pusher_state_min_timer,
+                                    profile->pusher_distance_min, profile->primary_braking_accel_min);
+}
+
+static int flightProfileCommand(size_t argc, char *argv[], size_t outbufc, char outbuf[]){
+  //Allows user to configure flight profiles
+  pod_mode_t mode = get_pod_mode();
+  if(mode == 1){ //Only allowed in Boot mode
```
Comment (EdHurtig) [inline core/commands.c]: This should be 

```
if (mode == Boot) {
```
Technically Boot Mode equates to `1` but the whole point of the `enum` is to disambiguate that

File: core/pod-defs.h [hunk 2]
```diff
@@ -147,6 +147,20 @@ typedef struct manual_config {
...
+    useconds_t watchdog_timer; // Main braking timeout initiated by pushing state // 5MPH 4300000
+    useconds_t emergency_hold; // time held in the emergency state
+    useconds_t braking_wait; // Time before engaging secondary brake, if needed
+    useconds_t braking_hold; // min time to hold brakes before vent
+    useconds_t pusher_timeout; // Timeout for the pusher plate debounce
+    float pusher_state_accel_min; // m/s/s Threshold for transitioning into the pushing state
+    useconds_t pusher_state_min_timer;  // Minimium time in the pushing state
+    float pusher_distance_min; // mm Distance to register the pusher as present
+    float primary_braking_accel_min; // m/s/s min acceptable acceleration while braking
+    pthread_rwlock_t lock;
```
Comment (EdHurtig) [inline core/pod-defs.h]: ❤️

**Review by EdHurtig — APPROVED** [review]

File: core/pod-defs.h [hunk 3]
```diff
@@ -410,6 +421,111 @@ static inline void set_value_f(pod_value_t *pod_field, float newValue) {
   pthread_rwlock_unlock(&(pod_field->lock));
 }
 
+//Setters for flight_profile
+static inline void set_watchdog_timer(flight_profile_t *profile, useconds_t newValue){
```
Comment (EdHurtig) [inline core/pod-defs.h:426]: `static inline` thank you for reusing and sticking to this :)

File: core/commands.c [hunk 4]
```diff
@@ -126,28 +137,32 @@ static int ventCommand(size_t argc, char *argv[], size_t outbufc, char outbuf[])
...
   if (start_hp_fill()) {
     return snprintf(outbuf, outbufc, "Entered HP Fill State");
   } else {
     return snprintf(outbuf, outbufc, "HP Fill Pre-Check Failure");
   }
 }
 
-static int standbyCommand(size_t argc, char *argv[], size_t outbufc, char outbuf[]) {
+static int standbyCommand(size_t argc, char *argv[], size_t outbufc,
+                          char outbuf[]) {
```
Comment (EdHurtig) [inline core/commands.c:156]: Interesting it seems that I've been guilty of some style violations... I'll have to check my formatting config

**Review by EdHurtig — APPROVED** [review]

**Comment by EdHurtig** [comment]
@shorstman mash that merge button when ready

**Comment by shorstman** [comment]
I am not authorized to do that

**Comment by EdHurtig** [comment]
Mmmmmm really?

**Pull request merged**